puzzlejs/puzzle.js

```diff
@@ -450,6 +463,88 @@ function PuzzleEntry(p, index) {
         }
     }
 
+    this.jumpToNearGrid = function(td, drow, dcol) {
+        let rcActive = this.activeGrid.grid.getBoundingClientRect();
+        let igridBest = -1;
+        let dzBest = 99999;
+        let puzzleGrid;
+
+        for (let igrid = 0; igrid < this.puzzleGrids.length; ++igrid) {
+            let grid, rc;
+            if ((puzzleGrid = this.puzzleGrids[igrid]) && (grid = puzzleGrid.grid) && (rc = grid.getBoundingClientRect())) {
+                let dz = this.dzDistanceToRect(rcActive, rc, drow, dcol);
+                if (dz >= 0 && dz < dzBest) { dzBest = dz; igridBest = igrid; }
+            }
+        }
+        if (igridBest == -1)
+            return false;
+
+        rcActive = td.getBoundingClientRect(); // look relative to current cell
+        let xActive = (rcActive.left + rcActive.right) / 2;
+        let yActive = (rcActive.top + rcActive.bottom) / 2;
+        puzzleGrid = this.puzzleGrids[igridBest];
+        dzBest = 9999999;
+        let tdBest = null;
+        for (let cell in puzzleGrid.lookup) {
```

Comment:
Not everything in `lookup` is a grid cell (example: dots). This would be safer, and not much different:

```
puzzleGrid.container.querySelectorAll(".inner-cell").forEach(td => {
    let rc;
    // continue the rest of your code here
});
```